### DISCUSSION TITLE
"DIMENSIONS" not found error

### DISCUSSION THREAD

**Discussion by qinghe-gm** [modflowpy/flopy#1760] [Q&A] [answered]
Dear all, 

I' m using FloPy to run a modflow model. I've finished the exercise in FloPy tutorial and the code works good. But today I tried to build my own model using similar code (including the very basic time/spatial discretization packages), but got this error:

ERROR REPORT:

  1. Required block "DIMENSIONS" not found.  Found end of file instead.

I did some search on google and some pages say this bug could be because the spatial definition of model's rows/cols or aquifer layer is missing. But I did define such info in the discretization package, as:
`ddis = fp.mf6.ModflowGwfdis(gwf,`
                           `length_units = "METERS",`
                           `nlay = nlay,`
                           `nrow = nrow,`
                           `ncol = ncol,`
                           `delr = delr,`
                           `delc = delc,`
                           `top = top,`
                           `botm = botm)`

So I wonder what else reason could cause this bug?

**Comment by spaulins-usgs** [comment, marked as answer]
In addition to the DIS package, the TDIS package and the stress packages (CHD, DRN, EVT, GHB, RIV, WEL, ...) also have dimensions blocks.  A missing dimension block from any of these packages could potentially cause the error you are seeing.  More information about what package is causing this error may be written to the end of your mfsim.lst file.

**Reply by qinghe-gm** [reply]
Thank you for your explanation!
